packages/core/src/controllers/store.ts

```diff
@@ -1,30 +1,42 @@
 import { generateChildLogger, getLoggerContext } from "@walletconnect/logger";
-import { ICore, IStore, PairingTypes, ProposalTypes, SessionTypes } from "@walletconnect/types";
+import { ICore, IStore } from "@walletconnect/types";
 import { getInternalError, isProposalStruct, isSessionStruct } from "@walletconnect/utils";
 import { Logger } from "pino";
 import { CORE_STORAGE_PREFIX, STORE_STORAGE_VERSION } from "../constants";
 import isEqual from "lodash.isequal";
 
-type StoreStruct = SessionTypes.Struct | PairingTypes.Struct | ProposalTypes.Struct;
-
-export class Store<Key, Data extends StoreStruct> extends IStore<Key, Data> {
+export class Store<Key, Data extends Record<string, any>> extends IStore<Key, Data> {
   public map = new Map<Key, Data>();
   public version = STORE_STORAGE_VERSION;
 
   private cached: Data[] = [];
   private initialized = false;
 
+  /**
+   * Regenerates the value key to retrieve it from cache
+   */
+  private getKey: ((data: Data) => Key) | undefined;
+
   private storagePrefix = CORE_STORAGE_PREFIX;
 
+  /**
```

Comment:
Felt like it made sense specifically because `getKey` is not immediately straightforward.

Maybe I can remove the JSDoc and change the function's name to `getKeyForCache` or something, what do you think

Reply:
Yeah assuming the type inference is meaningful enough when inspecting the constructor param, I think we can go with `getKeyForCache` or `getKeyForCacheRestore`.

If you feel like it's a lot better with the JSDoc comment feel free to leave it though 👍

Reply:
I personally feel like it makes sense to leave JSDoc in, maybe in the future `getKey` can be expanded on and used for other reasons (eg: Logging) so just adding some documentation doesn't hurt anyone I think. But I do not feel _very_ strongly about this.

Reply:
Ok let's leave it there :)